packages/flutter_deck/lib/src/controls/fullscreen/fullscreen.dart

```diff
@@ -1,25 +1,45 @@
-export 'stub.dart'
-    if (dart.library.io) 'io.dart'
-    if (dart.library.html) 'web.dart';
+import 'package:flutter_deck/src/controls/fullscreen/window_proxy/window_proxy.dart';
 
 /// Describes a collection of functions that can mutate and retrieve
 /// the application's fullscreen state.
-abstract class FlutterDeckFullscreenManagerBase {
+class FlutterDeckFullscreenManager {
+  /// Default constructor for [FlutterDeckFullscreenManager].
+  ///
+  /// The [windowProxy] is required for getting and setting window properties
+  /// from underlying platform.
+  ///
+  FlutterDeckFullscreenManager(WindowProxyBase windowProxy)
+      : _windowProxy = windowProxy;
+
+  final WindowProxyBase _windowProxy;
+  bool _initialized = false;
+
+  Future<WindowProxyBase> get _instance async {
+    if (!_initialized) {
+      await _windowProxy.initialize();
+      _initialized = true;
+    }
+    return _windowProxy;
+  }
+
   /// Returns whether or not this platform has support to enter/leave fullscreen
-  bool canFullscreen() => false;
+  bool canFullscreen() => _windowProxy.canFullscreen();
```

Comment:
Just to clarify: why are you calling the `_windowProxy` directly here and not the `_instance`?

Reply:
Correct. Also instance is a future, therefore we would need to make this `Future<bool>`, and that is not needed